dolphin/api/v1/storages.py

```diff
@@ -139,22 +140,23 @@ def sync(self, req, id):
         :return:
         """
         # validate the id
-        context = req.environ.get('dolphin.context')
-        # admin_context = context.RequestContext('admin', 'fake', True)
+        ctxt = req.environ['dolphin.context']
         try:
-            device = db.access_info_get(context, id)
-        except Exception as e:
+            storage = db.storage_get(ctxt, id)
+        except exception.StorageNotFound as e:
             LOG.error(e)
-            raise exception.AccessInfoNotFound(e)
-
-        tasks = (
-            'pool_task',
-            'volume_task'
-        )
-        for task in tasks:
-            self.task_rpcapi.sync_storage_resource(context, id, task)
-
-        return dict(name="Sync storage 1")
+            raise exc.HTTPBadRequest(explanation=e.message)
```

Comment:
use e.msg here. reason: if you see the DolphinException __init__, the actual message is stored in self.msg

Reply:
done